src/network/protocol/messages/notfound.zig

```diff
@@ -0,0 +1,127 @@
+const std = @import("std");
+const protocol = @import("../lib.zig");
+const Sha256 = std.crypto.hash.sha2.Sha256;
+const InventoryVector = @import("lib.zig").InventoryVector;
+
+/// NotFoundMessage represents the "notfound" message
+///
+/// https://developer.bitcoin.org/reference/p2p_networking.html#notfound
+pub const NotFoundMessage = struct {
+    inventory: []const InventoryVector,
+
+    const Self = @This();
+
+    pub fn name() *const [12]u8 {
+        return protocol.CommandNames.NOTFOUND ++ [_]u8{0} ** 4;
+    }
+
+    /// Returns the message checksum
+    ///
+    /// Computed as `Sha256(Sha256(self.serialize()))[0..4]`
+    pub fn checksum(self: *const Self) [4]u8 {
+        var digest: [32]u8 = undefined;
+        var hasher = Sha256.init(.{});
+        const writer = hasher.writer();
+        self.serializeToWriter(writer) catch unreachable; // Sha256.write is infallible
+        hasher.final(&digest);
+
+        Sha256.hash(&digest, &digest, .{});
+
+        return digest[0..4].*;
+    }
+
+    /// Serialize a message as bytes and write them to the buffer.
+    ///
+    /// buffer.len must be >= than self.hintSerializedLen()
+    pub fn serializeToSlice(self: *const Self, buffer: []u8) !void {
+        var fbs = std.io.fixedBufferStream(buffer);
+        try self.serializeToWriter(fbs.writer());
+    }
+
+    /// Serialize the message as bytes and write them to the Writer.
+    pub fn serializeToWriter(self: *const Self, writer: anytype) !void {
+        try writer.writeInt(u32, @intCast(self.inventory.len), .little);
+        for (self.inventory) |inv| {
+            try writer.writeInt(u32, inv.type, .little);
+            try writer.writeAll(&inv.hash);
+        }
```

Comment:
If it will be used by other people, you should move this as it's own function

Reply:
same for deser

Reply:
Okay boss I will move it

Reply:
Hey boss what should I move?

Reply:
@tdelabro I have moved the shared function to lib.zig